Automattic/jetpack
# Add Admin menu generic class

Required by D59042-code.

#### Changes proposed in this Pull Request:

Our current admin menu class contains logic that's not generic enough for other cases (e.g. domain-only sites). On this PR we are adding a truly generic class that can be extended for any other scenario since it only contains the common logic that's always needed.

#### Jetpack product discussion


N/A
#### Does this pull request change what data or activity we track or use?


No
#### Testing instructions:






This PR does not changes the current behavior. Do a smoke test across multiple sites and make sure that there are no regressions.

_Simple_
- Apply  D59038-code to your WP.com sandbox.
- Sandbox the API and a simple site.
- Go to https://wordpress.com and switch to the sandboxed site.
- Check the menu under different scenarios (Calypso, WP Admin, toggle on/off) and make sure there are no regressions.

_Atomic_
- Install Jetpack Beta on a Atomic site and switch to the branch of this PR.
- Go to https://wordpress.com and switch to the WoA site.
- Check the menu under different scenarios (Calypso, WP Admin, toggle on/off) and make sure there are no regressions.

_Jetpack_
- Spin up a Jetpack site running this branch.
- Spin up a local instance of Calypso with [this code](https://github.com/Automattic/wp-calypso/blob/46541a684523325e4efd293d0f8b7f2530f4e6f2/client/state/selectors/is-nav-unification-enabled.js#L27-L31) removed/commented out.
- Go to http://calypso.localhost:3000 and switch to the Jetpack site.
- Check the menu in Calypso only (since Nav Unification is disabled on WP Admin) and make sure there are no regressions (all menus should point to Calypso regardless of the toggle, except Feedback, Customize, Import, Export and WP Admin).

Additionally, checks how this new class is used in D59042-code.
